noxfile.py

```diff
@@ -429,44 +452,37 @@ def set_minimal_package_versions(session: nox.Session):
                 external=True,
             )
 
-    _run_cargo_set_package_version(session, "indexmap", "1.6.2")
-    _run_cargo_set_package_version(session, "hashbrown:0.12.3", "0.9.1")
-    _run_cargo_set_package_version(session, "plotters", "0.3.1")
-    _run_cargo_set_package_version(session, "plotters-svg", "0.3.1")
-    _run_cargo_set_package_version(session, "plotters-backend", "0.3.2")
-    _run_cargo_set_package_version(session, "bumpalo", "3.10.0")
-    _run_cargo_set_package_version(session, "once_cell", "1.14.0")
-    _run_cargo_set_package_version(session, "rayon", "1.5.3")
-    _run_cargo_set_package_version(session, "rayon-core", "1.9.3")
-
-    # string_cache 0.8.4 depends on parking_lot 0.12
-    _run_cargo_set_package_version(session, "string_cache:0.8.4", "0.8.3")
-
-    # 1.15.0 depends on hermit-abi 0.2.6 which has edition 2021 and breaks 1.48.0
-    _run_cargo_set_package_version(session, "num_cpus", "1.14.0")
-    _run_cargo_set_package_version(
-        session, "num_cpus", "1.14.0", project="examples/word-count"
-    )
-
-    projects = (
-        None,
-        "examples/decorator",
-        "examples/maturin-starter",
-        "examples/setuptools-rust-starter",
-        "examples/word-count",
-    )
     for project in projects:
-        _run_cargo_set_package_version(
-            session, "parking_lot:0.12.1", "0.11.0", project=project
-        )
-        _run_cargo_set_package_version(session, "once_cell", "1.14.0", project=project)
-
-    _run_cargo_set_package_version(
-        session, "rayon", "1.5.3", project="examples/word-count"
-    )
-    _run_cargo_set_package_version(
-        session, "rayon-core", "1.9.3", project="examples/word-count"
-    )
+        lock_file = "Cargo.lock"
+        if project:
+            lock_file = os.path.join(project, "Cargo.lock")
```

Comment:
We already import `Path` above, so I think you could simplify this a bit like

```python
lock_file = Path(project) / "Cargo.lock"
```

and

```python
cargo_lock = toml.loads(lock_file.read_text())
```